src/libDirectoryService/DirectoryService.cpp

```diff
@@ -257,6 +257,10 @@ bool DirectoryService::ProcessSetPrimary(const bytes& message,
     m_consensusMyID++;
   }
 
+  // Add ds guard to exclude list for ds comm at bootstrap
+  Guard::GetInstance().AddDSGuardToBlacklistExcludeList(
+      *m_mediator.m_DSCommittee);
```

Comment:
Maybe this piece of code should also be added inside Node::StartRetrieveHistory(), after setting the dscommittee from persistence? (https://github.com/Zilliqa/Zilliqa/blob/master/src/libNode/Node.cpp#L485)

Reply:
already exists in the branch

Reply:
I saw the code now